### ISSUE TITLE
False negative with Protocol and abstract methods

### ISSUE THREAD

**Issue by vtgn** [python/mypy#18034]
**Bug Report**

Hello. I just try to use the Protocol feature of Python, and I remark several false negative with mypy with the static methods.

**To Reproduce**
 https://mypy-play.net/?mypy=latest&python=3.12&gist=28822a92ae617fb3a81e17cdf01eb894

```py
class A(Protocol):
    @staticmethod
    def static_meth() -> int: ...

    def meth(self) -> int: ...


class B(A):
    def func(self) -> int:
        return B.static_meth() + self.meth() # <== Calling B.static_meth() should be reported as an error, because it returns None during execution, which is an invalid type here !! Pylance does, not about the NoneType of this call, but because it says the static method is abstract and unimplemented, which is a better reason in my opinion.


class C(B):
    @staticmethod
    def static_meth() -> int:
        return 1

    def meth(self) -> int:
        return 2


b = B() # <== this is the only error reported by mypy on this entire script, which is correct, Pylance reports it too : Cannot instantiate abstract class "B" with abstract attributes "meth" and "static_meth"mypy[abstract](https://mypy.readthedocs.io/en/stable/_refs.html#code-abstract). Notice how the error message says the static_meth is abstract here, but not when it is called on B or b.

c = C()

print(c.func()) # <== Causes an error during execution that should have been avoided if mypy had reported the invalid call to B.static_meth() in the implementation of func() : TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

print(B.static_meth()) # <== No error detected here, calling B.static_meth() should be reported as an error because it is not implemented! Only Pylance does.

print(b.static_meth()) # <== No error detected here, calling b.static_meth() should be reported as an error because it is not implemented! Pylance does not either.
```

**Expected Behavior**
mypy should report all the calls to `B.static_meth()` and `b.static_meth()` because `static_meth()` is not implemented in the B class or in one of its ancestors.

**Actual Behavior**
mypy doesn't report the calls to unimplemented static methods on classes and instances, although it says exactly this when you try to instantiate the class. Pylance does, but not in all the expected cases. :/

**Your Environment**

- Mypy version used: 1.11.2 (still present on the latest version 1.13, as you can see with the link to mypy playground above)
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.12.6

Regards.
